### PR TITLE
Fix `gr.SelectData` so that the target attribute is correctly attached, and the filedata is included in the data attribute with `gr.Gallery`

### PR DESCRIPTION
**Pull request by abidlabs**
Fixes: #5758

Test with:

```py
import gradio as gr

with gr.Blocks() as demo:
    table = gr.Dataframe([[1, 2, 3], [4, 5, 6]])
    textbox = gr.Textbox("Hello World!")

    statement = gr.Textbox()

    def on_select(evt: gr.SelectData):  # SelectData is a subclass of EventData
        return f"You selected {evt.value} at {evt.index} from {evt.target}"

    table.select(on_select, None, statement)
    textbox.select(on_select, None, statement)

demo.launch()
```

### CODE REVIEW

File: gradio/route_utils.py [hunk 1]
```diff
@@ -186,22 +187,30 @@ def restore_session_state(app: App, body: PredictBody):
     return session_state, iterators
 
 
+def prepare_event_data(
```
Comment (abidlabs) [inline gradio/route_utils.py:190]: Extracted this as a separate method to make it easier to test

File: js/gallery/static/Gallery.svelte [hunk 2]
```diff
@@ -131,7 +131,7 @@
 			if (selected_image !== null) {
 				dispatch("select", {
 					index: selected_image,
-					value: _value?.[selected_image][1]
+					value: [_value?.[selected_image][0].data, _value?.[selected_image][1]]
```
Comment (abidlabs) [inline js/gallery/static/Gallery.svelte:134]: Before, the value of the selection was just the caption, which was usually `None`. This changes it to be [image_data, caption], where the image_data is usually a filepath/url.

**Comment by abidlabs** [comment]
Let me fix the conflicts real quick

**Review by dawoodkhan82 — APPROVED** [review]
Looks good, also tested with chatbot. Works great

**Comment by abidlabs** [comment]
Thanks @dawoodkhan82! I'll fix the conflicts and merge in

**Pull request merged**